### ISSUE TITLE
[Feature request] 2 request

### ISSUE THREAD

**Issue by LLKoder** [sunnyark/civitai-shortcut#21]
1. It would be nice if models have to custom directory path and custom filename to download.
2. It would be nice if main thumbnails in extra networks will be able to change by the shortcut extension. I mean to have a button near each images and it will change the main thumbnail.

**Comment by sunnyark** [comment]
@LLKoder Thanks @LLKoder for the issue. 
I will do my best to incorporate your feedback. Thank you.

**Comment by LLKoder** [comment]
@sunnyark 
As I understand you realised function of custom directory and file names in last commit https://github.com/sunnyark/civitai-shortcut/commit/cfd5c9adcc7d93d1d7afae4c72ba54c0a3efee7e. Where can I see that function if I right undestand.

Sorry I don't know Korean language and translator do not work very well.

**Comment by sunnyark** [comment]
@sunnyark The recent update includes a feature that recommends suitable folder names for users. Although custom folder functionality was implemented in the previous update, it may not be fully satisfactory. When downloading, if you select "Create model name folder" from the dropdown menu, a text box for modifying the folder name will appear on the right. If you choose "Create individual version folder" at the bottom, you can create separate folders for each version below. Unfortunately, the ability to modify file names has not been implemented. Modifying file names would make it difficult to provide accurate information as there are many interconnected components associated with the file names. If you wish, I recommend manually modifying the folder by using the "Open download folder" button. When making modifications, please ensure that the names of the three files, "info", "preview", and "model file", are aligned.

**Comment by sunnyark** [comment]
@LLKoder If you change the file name, the extension program will consider that you have not downloaded the corresponding model. However, since the extension program itself does not perform any actions with the model, it will simply result in a mismatch of information. The role of the extension program is primarily to manage the shortcuts for Civitai, and it minimally handles the management of downloaded models.

Regarding the folder structure for downloads, by default, it is organized with a hierarchy of the type base folder, followed by the model name folder, and then the individual version folder. For example, in the case of the "lora" type, the structure would be:

models/lora/model_name/individual version folder

You are free to choose the names for both the model_name and individual version folder as you desire. Additionally, you can create separate folders based on classification names, but these folders cannot have subfolders. The purpose of these classification folders is to simplify the representation of complex folder structures in the extra network by limiting the folder names.

As mentioned before, the role of the extension program is to manage shortcuts for Civitai and does not actively manage the downloaded models. Therefore, if you choose to move the downloaded files or change their locations within the download folder, it will not cause any issues with the extension program itself. It will simply result in a minor discrepancy in the information. However, as long as the downloaded files are located within the type base folder, you can roughly manage them through the "Downloaded Model Information" tab.

I'm not a native English speaker, so I'm using the help of GPT. Please excuse any awkwardness in my language

**Comment by LLKoder** [comment]
I see. Thanks a lot. :)

**Comment by LLKoder** [comment]
![2023-05-29_12-29](https://github.com/sunnyark/civitai-shortcut/assets/50442676/dc693560-aa45-49fa-9514-13f572c618a8)

This button not work, it not change preview in the extension and not change  the preview in extra network.
Or I misunderstand what it means.

**Comment by sunnyark** [comment]
@LLKoder If you refresh on the extra network, it should be reflected.

**Comment by LLKoder** [comment]
@sunnyark 
I see, it's not working for me because my model name is not default and extension put file with default name. :(

`tifkeller.preview.png` - file is putting by extension.
`JernauGurgeh-Tiffany_Keller.pt` - my model name.

Although, my directory is not default too, but the extension find my directory correctly.

**Comment by sunnyark** [comment]
@LLKoder Yes, for that reason, it is not possible to rename the file when downloading. If file name changes are allowed, there are difficulties in continuously tracking and managing the file. However, folder name changes can be done without any issues. If you have saved it in multiple places, it may work somewhat inaccurately.

**Comment by sunnyark** [comment]
@LLKoder I will try to find a way to ensure that the preview image is correctly saved even if the file name has been changed.

**Comment by sunnyark** [comment]
@LLKoder Thank you. There are some delays due to a few tests. If everything is fine, I will proceed with the upload.
The ongoing work is being uploaded to the working branch, so if you want to know the progress, please take a look around.
